`CI` `Send notification (PR)` fails for PRs from forks

https://github.com/Hammerspoon/hammerspoon/actions/runs/4283478942/jobs/7459108879#step:13:7
```
Run /usr/bin/curl -H "X-CI-WebHook: true" -H "Content-Type: application/json" -d '{"secret": "", "repository": "hammerspoon", "workflow": "CI (PR)", "message": "CI for https://github.com/Hammerspoon/hammerspoon/pull/3353 (cmsj)\n  * Spelling\n  * 	 Executed 3[1](https://github.com/Hammerspoon/hammerspoon/actions/runs/4283478942/jobs/7459108879#step:13:1)2 tests, with 52 tests skipped and 0 failures (0 unexpected) in 3773.852 (3774.2[6](https://github.com/Hammerspoon/hammerspoon/actions/runs/4283478942/jobs/7459108879#step:13:7)2) seconds"}' 
curl: no URL specified!
curl: try 'curl --help' or 'curl --manual' for more information
Error: Process completed with exit code 2.
```

https://github.com/Hammerspoon/hammerspoon/blob/82d215936801198e03dc7e25f2aa8242c5e02965/.github/workflows/ci_testbuild.yml#L74

The code expects `${{ secrets.CI_WEBHOOK_URL }}` to have a value, but for `pull_request` (not `pull_request_target`) from a forked repository, it won't for security reasons. 

It's possible (but not recommended) to make this work by splitting things into two jobs and setting up a restricted permission token for the first job, but i think the simpler to reason through fix is to just condition the step on having a secret.

--- a/.github/workflows/ci_testbuild.yml
+++ b/.github/workflows/ci_testbuild.yml
@@ -69,6 +69,8 @@ jobs:
           /usr/bin/curl -H "X-CI-WebHook: true" -H "Content-Type: application/json" -d '{"secret": "${{ secrets.CI_WEBHOOK_SECRET }}", "repository": "hammerspoon", "workflow": "CI (Push)", "message": "CI for ${{ github.event.compare }} (${{ github.event.head_commit.author.name }}):\n  * ${{ github.event.head_commit.message}}\n  * ${{ steps.test.outputs.test_result }}"}' ${{ secrets.CI_WEBHOOK_URL }}
 
       - name: Send notification (PR)
-        if: ${{ always() && github.event_name == 'pull_request' && github.repository == 'hammerspoon/hammerspoon' }}
+        if: ${{ always() && github.event_name == 'pull_request' && github.repository == 'hammerspoon/hammerspoon' && env.CI_WEBHOOK_URL != '' }}
+        env:
+          CI_WEBHOOK_URL: ${{ secrets.CI_WEBHOOK_URL }}
         run: |
-          /usr/bin/curl -H "X-CI-WebHook: true" -H "Content-Type: application/json" -d '{"secret": "${{ secrets.CI_WEBHOOK_SECRET }}", "repository": "hammerspoon", "workflow": "CI (PR)", "message": "CI for ${{ github.event.pull_request.html_url }} (${{ github.actor }})\n  * ${{ github.event.pull_request.title }}\n  * ${{ steps.test.outputs.test_result }}"}' ${{ secrets.CI_WEBHOOK_URL }}
+          /usr/bin/curl -H "X-CI-WebHook: true" -H "Content-Type: application/json" -d '{"secret": "${{ secrets.CI_WEBHOOK_SECRET }}", "repository": "hammerspoon", "workflow": "CI (PR)", "message": "CI for ${{ github.event.pull_request.html_url }} (${{ github.actor }})\n  * ${{ github.event.pull_request.title }}\n  * ${{ steps.test.outputs.test_result }}"}' $CI_WEBHOOK_URL

